source/client-side-encryption/tests/README.rst

```diff
@@ -0,0 +1,220 @@
+============================
+Client Side Encryption Tests
+============================
+
+.. contents::
+
+----
+
+Introduction
+============
+
+This document describes the format of the driver spec tests included in the JSON
+and YAML files included in this directory.
+
+Additional prose tests, that are not represented in the spec tests, are described
+and MUST be implemented by all drivers.
+
+TODO
+----
+Once limitations of query parsing in mongocryptd are known, add tests to exercise those limitations.
+
+Spec Test Format
+================
+
+The spec tests are closely based on the transactions spec tests and CRUD spec tests, with the following notable differences:
+
+- An ``encryption_setup`` section, describing client side encryption setup.
+
+- The addition of ``clientSideEncryptionOptions`` on clients.
+
+- The addition of the per-test ``kmsFailpoint`` and ``mongocryptdFailpoint`` to control the mock KMS an mock mongocryptd servers.
+
+- Operations may be specified per collection through an optional "object" field. I.e. operations use the default collection specified at the root of the test:
+
+  .. code:: json
+
+     {
+         "name": "insertOne",
+         "documents": [ { "_id": 1 } ]
+     }
+
+  inserts on the default collection, but
+
+  .. code:: json
+
+      {
+          "name": "insertOne",
+          "object": { "database": "db", "collection": "other" }
+          "documents": [ { "_id": 1 }]
+      }
+
+  inserts in the ``db.other`` collection.
+
+- The addition of the ``mongocryptd_expectations``, for APM events coming from mongocryptd.
+
+
+Each YAML file has the following keys:
```

Comment:
Is there any need to express server version/topology requirements? If so, can we pull in `runOn` syntax from the CRUD/transaction specs? If not, should the README above specify some criteria for deciding to run these tests (e.g. driver is built with FLE support enabled)?

Reply:
Sorry, I forgot that the CRUD spec wasn't updated yet (pending [SPEC-1255](https://jira.mongodb.org/browse/SPEC-1255)). I'm referring to the syntax in https://github.com/mongodb/specifications/blob/master/source/transactions/tests/README.rst#test-format.